simd/src/Kokkos_SIMD_Common_Math.hpp

```diff
@@ -97,18 +97,38 @@ template <
     std::enable_if_t<!std::is_same_v<Abi, simd_abi::scalar>, bool> = false>
 KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION T
 reduce(basic_simd<T, Abi> const& v,
-       typename basic_simd<T, Abi>::mask_type const& m, T identity,
-       BinaryOperation op = {}) {
+       typename basic_simd<T, Abi>::mask_type const& m, BinaryOperation op = {},
+       T identity = Impl::Identity<T, BinaryOperation>()) {
   if (none_of(m)) {
     return identity;
   }
-  T result = Impl::Identity<T, BinaryOperation>();
+  bool init = true;
+  T result  = Impl::Identity<T, BinaryOperation>();
```

Comment:
We need to modify this, otherwise we don't take into account the user provided identity.
```suggestion
  T result  = identity;
```

Reply:
Yes, in fact, this is the bug that I meant to address originally for this PR, but was sidetracked while considering to enforce the constraint on the `identity_element`.
Fixed to use `identity` argument.